scripts/util.py

```diff
@@ -51,8 +64,74 @@ def read_source_root_path():
 def read_proof_root_path():
     return read_path_from_stdin("the 'proofs' directory (usually '.')")
 
-def read_litani_path():
-    return read_path_from_stdin("the litani script")
+def is_litani_submodule_in_path(path):
+    is_in_path = False
+    try:
+        sys.path.insert(1, path)
+        importlib.import_module("litani")
+        is_in_path = True
+    except ModuleNotFoundError:
+        print(f"Litani submodule was not found at directory: {path})")
+    del sys.path[1]
+    return is_in_path
+
+def add_litani_submodule(path):
+    """Adds the litani submodule to the specified path"""
+    try:
+        subprocess.run(
+             ["git", "submodule", "add", "--quiet", "--name", "litani", "--",
+             "https://github.com/awslabs/aws-build-accumulator.git", "litani"],
+             check=True, cwd=path)
+    except subprocess.CalledProcessError:
+        print(f"Could not add Litani submodule at path: {path}")
+
+def is_litani_installed():
+    return shutil.which("litani") is not None
+
+def determine_litani_origin():
+    choice = None
+    while True:
+        print(LITANI_DIALOG, file=sys.stderr)
+        choice = read_from_stdin()
+        if choice in ("1", "2", "3"):
+            break
+        print("You have entered an invalid value. Please choose again.")
+    return choice
+
+def install_litani_brew_package():
+    """Runs `brew install litani`. Exits if a failure occurs"""
+    try:
+        proc = subprocess.run(
+            ["brew", "install", "litani"],
+            check=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if proc.returncode:
+            logging.warning(proc.stderr.strip().decode())
+            sys.exit(1)
+    except subprocess.CalledProcessError as e:
```

Comment:
This isn't reachable because you're doing `check=False` I think?

What I actually suggest is:
1. `check=True`
2. Don't bother with `if proc.returncode`
3. Catch both `CalledProcessError`, and also `FileNotFoundError` (I think this gets raised if `brew` is not on the PATH or not executable) and handle the error there. If brew fails I think its own stderr should be sufficient, if it's not on the PATH then logging a message would probably helpful.

Reply:
wrt 3.:

We indeed have to handle for `FileNotFoundError` as well. Thanks for pointing that out.

Informative messages regarding the underlying error will be logged out even if we just log the error itself.

For example, if I change this function to instead attempt to install "slitani" via brew (intended typographical error) then I get:

```
Warning: No available formula with the name "slitani". Did you mean litani?
Error: No previously deleted formula found.
==> Searching taps on GitHub...
Error: No formulae found in taps.
WARNING:root:Failed in running `brew install litani`
WARNING:root:Command '['brew', 'install', 'slitani']' returned non-zero exit status 1.
```

Additionally, if I update my `PATH` and make `brew` temporarily unavailable then calling the soon-to-be-revised function that logs the error then you would see:

```
WARNING:root:`brew` is missing from your PATH
WARNING:root:[Errno 2] No such file or directory: 'brew'
```

which itself should be informative enough.